### ISSUE TITLE
git-ultrasphinx won't start because of insufficient permissions

### ISSUE THREAD

**Issue by christiaan** [rosenfeld/gitorious-cookbooks#12]
First it complained about being unable to write the pid file to '/var/www/gitorious/db/sphinx/log/searchd.pid'

After doing a chmod -R 0777 on /var/www/gitorious/db/sphinx/log/
It complained about being unable to open the sphinx_index_main.spl file and ended with "FATAL: no valid indexes to serve"

Then I did a chmod -R 0777 of /var/www/gitorious/db/sphinx/
And now it finally starts.

This happened on boot and when trying to execute /etc/init.d/git-ultrasphinx start

**Comment by christiaan** [comment]
I just want to note that besides this single simple to fix error everything seems to work flawless on a fresh Debian 6.0.2 net install.

**Comment by rosenfeld** [comment]
Yes, I can understand that the way how database creation is currently handled is error prone because it does too much:

https://github.com/rosenfeld/gitorious-cookbooks/blob/e02e3f5779c218775f4fc46ad3634fa79379d2bc/gitorious/recipes/default.rb

<pre>
script "create gitorious database" do
  interpreter "bash"
  cwd deploy_path
  code %Q{
    mysqladmin create #{db_database}
    mysql -e "GRANT ALL ON #{db_database}.* TO '#{db_user}'@'localhost' IDENTIFIED BY '#{db_password}';"
    export RAILS_ENV=production
    bundle exec rake db:setup
    echo -e "#{node[:gitorious][:admin][:email]}\\n#{node[:gitorious][:admin][:password]}" | script/create_admin
    chown -R git:git .
  }
  not_if "mysql -e 'show databases' | grep -q #{db_database}"
end
</pre>


This can leave the installation in a bad state if some of the commands fail while others succeed. In your case, the last command ("chown -R git:git .") is a workaround for avoiding the problem you had and making the recipe easier to write and understand.

But making it less error prone would probably make this recipe harder to understand while the chances of that happening, like it did with you, are low. So, I'm not going to invest much time on this, unless it always happen, but I'm accepting any pull request fixing the database creation and/or the ultrasphinx multiple-tasks-in-a-shell-workaround issue.

But as I don't have much free time, I prefer to use it to things I appreciate more like moving Gitorious to Rails 3 and help improving it. This recipe was an attempt to make the install process less painful, but I don't mind if occasionally someone still needs to do some manual adjustments because still most of the work will get done by the recipe.

Anyone willing to contribute to this recipe is free to send patches or pull requests and I'll gladly apply/merge them.

**Comment by christiaan** [comment]
I did a complete reinstall including Debian 6.0.2 and it resulted in this same error.
Maybe it is an idea to open up the Wiki here on github and have a list with know issues per debian version?

Instead of the chmod I now executed "chown -R git:git /var/www/gitorious/db/sphinx" which also solves the problem.

**Comment by rosenfeld** [comment]
Ok, if the problem is happening always, I'll try to fix it this weekend. I'll download the 6.0.2-1 (latest) Debian CD image.

I'll open the Wiki, but I won't maintain it :)

Thanks for taking time to test its reprodutibility and for opening this issue.

**Comment by rosenfeld** [comment]
I've just tried on latest Debian and didn't have any issues (except by some temporary network problems which running chef-solo again when that happened continued normally).

So, sorry, I can't reproduce this problem. Also, I've changed the json.node example to point to v2.0.0 branch.

**Comment by adayth** [comment]
I had the same issue with Debian 6.0.3 i386 netinstall. Executing "chown -R git:git /var/www/gitorious/db/sphinx" solved it.

**Comment by rosenfeld** [comment]
The problem is that I couldn't reproduce this issue yet or understand what has caused it... It seems some of the tasks in the recipe aren't completely running, generating a side-effect, but it is a bit complicate to fix something that I don't completely understand or can reproduce... I'm sure it is possible to write the tasks in a less error-prone way, but that would also mean that the recipe would get much more complicated to read and maintain, so there are trade-offs... I would rather prefer not to change it since this issue is not a common one and is not reproducible and the fix is already documented in this issue...

**Comment by rosenfeld** [comment]
Please, take a look at my first comment in this issue and let me know if you have some idea to properly write that part. It seems that either "rake db:setup" or "script/create_admin" is failing... Maybe that would prevent the chown to run, but I'm just guessing here... But if just running the chown fixed the issue for you with no extra procedures, than it is likely that both rake and create_admin succeeded. So, maybe a workaround would be to just add a task for chown the directory after each run...

**Comment by rosenfeld** [comment]
Also, maybe the problem is not in the database creation task, but in the ultrasphinx setup task:

<pre>
script "setup ultrasphinx for Gitorious" do
  interpreter "bash"
  cwd         deploy_path
  code %Q{
    export RAILS_ENV=production
    bundle exec rake ultrasphinx:bootstrap

    aspell config dict-dir /usr/lib/aspell
    cp vendor/plugins/ultrasphinx/examples/ap.multi /usr/lib/aspell/
    bundle exec rake ultrasphinx:spelling:build

    chown git:git config/ultrasphinx/production.conf
  }
  creates     "#{deploy_path}/config/ultrasphinx/production.conf"
  notifies    :restart, "service[apache2]"
end
</pre>


Maybe, something went wrong before the chown, preventing it from running... As I've told you, it is not easy to understand what could be possibly raising this issue...
